base/common/src/main/java/com/netscape/certsrv/profile/ProfileData.java

```diff
@@ -273,18 +286,339 @@ public boolean equals(Object obj) {
+            profileData.setAuthenticatorId(value);
+        }
+        NodeList authzAclList = profileDataElement.getElementsByTagName("authzAcl");
+        if (authzAclList.getLength() > 0) {
+            String value = authzAclList.item(0).getTextContent();
+            profileData.setAuthzAcl(value);
+        }
+        NodeList renewalList = profileDataElement.getElementsByTagName("renewal");
+        if (renewalList.getLength() > 0) {
+            String value = renewalList.item(0).getTextContent();
+            profileData.setRenewal(Boolean.valueOf(value));
+        }
+        NodeList xmlOutputList = profileDataElement.getElementsByTagName("xmlOutput");
+        if (xmlOutputList.getLength() > 0) {
+            String value = xmlOutputList.item(0).getTextContent();
+            profileData.setXMLOutput(Boolean.valueOf(value));
+        }
+        NodeList profileInputList = profileDataElement.getElementsByTagName("Input");
+        int piCount = profileInputList.getLength();
+        for (int i = 0; i < piCount; i++) {
+            Element piElement = (Element) profileInputList.item(i);
+            ProfileInput profileInput = ProfileInput.fromDOM(piElement);
+            profileData.addProfileInput(profileInput);
+        }
+        NodeList profileOutputList = profileDataElement.getElementsByTagName("Output");
+        int poCount = profileOutputList.getLength();
+        for (int i = 0; i < poCount; i++) {
+            Element poElement = (Element) profileOutputList.item(i);
+            ProfileOutput profileOutput = ProfileOutput.fromDOM(poElement);
+            profileData.addProfileOutput(profileOutput);
+        }
+
+        NodeList policySetList = profileDataElement.getElementsByTagName("PolicySet");
+        int policySetCount = policySetList.getLength();
+        for (int i = 0; i < policySetCount; i++) {
+            Vector<ProfilePolicy> value = new Vector<>();
+            ProfilePolicy profilePolicy = new ProfilePolicy();
+            NodeList ppList = profileDataElement.getElementsByTagName("def");
+            if (ppList.getLength() > 0) {
+                Element ppElement = (Element) ppList.item(0);
+                PolicyDefault pd = PolicyDefault.fromDOM(ppElement);
+                profilePolicy.setDef(pd);
+            }
+            NodeList constraintList = profileDataElement.getElementsByTagName("constraint");
+            if (constraintList.getLength() > 0) {
+                Element constraintElement = (Element) constraintList.item(0);
+                PolicyConstraint pc = PolicyConstraint.fromDOM(constraintElement);
+                profilePolicy.setConstraint(pc);
+            }
+            NodeList valueIdList = profileDataElement.getElementsByTagName("value");
+            Element valueIdElement = (Element) valueIdList.item(0);
+            String valueId = valueIdElement.getAttribute("id");
+            profilePolicy.setId(valueId);
+            value.add(profilePolicy);
```

Comment:
I think there could be multiple `<value>` elements (i.e. `ProfilePolicy`) for each `<PolicySet>`, so this should be in a loop.

Reply:
Agreed. This caused a bit of sadness as multiple sub-elements contain `<value>` elements, but the `<value>` element we want here has an `id` attribute so we can filter based on that.